q2_types/feature_data/tests/test_format.py

```diff
@@ -159,6 +159,27 @@ def test_dna_fasta_format_empty_file(self):
 
         format.validate()
 
+    def test_dna_fasta_format_consecutive_IDs(self):
+        filepath = self.get_data_path('dna-sequences-consecutive-ids.fasta')
+        format = DNAFASTAFormat(filepath, mode='r')
+
+        with self.assertRaisesRegex(ValidationError, 'consecutive IDs.*1'):
+            format.validate()
+
+    def test_dna_fasta_format_missing_initial_ID(self):
+        filepath = self.get_data_path('dna-sequences-first-line-not-id.fasta')
+        format = DNAFASTAFormat(filepath, mode='r')
+
+        with self.assertRaisesRegex(ValidationError, 'First line'):
+            format.validate()
+
+    def test_dna_fasta_format_corrupt_characters(self):
+        filepath = self.get_data_path('dna-sequences-corrupt-characters.fasta')
+        format = DNAFASTAFormat(filepath, mode='r')
+
+        with self.assertRaisesRegex(ValidationError, 'Unicode.*2'):
+            format.validate()
+
```

Comment:
For the sake of testing, is it worth editing `_check` to parameterize the buffer_size? That way we can test edge cases in the buffer_size handling without needing huge files of garbage?

This would probably look like wrapping an `open(..., 'rb', buffer_size=...)` like described above, and then by default that is always io.DEFAULT_BUFFER_SIZE (but set to something much smaller while testing).

Reply:
I considered doing this, but I forgot to ask you. Are special parameters for testing purposes generally acceptable?

Reply:
It's definitely odd, but this seems unusual enough of a testing situation to warrant it. I think the benefits outweigh the strangeness. Also, we could conceptually consider it "part of the API" which we just so happen to never use. It's not an unreasonable parameter to have, so it's not as "fake" as it could be.

Reply:
I'm running into an issue I ran into before where the optional "buffering" parameter for the open() function doesn't seem to be actually doing anything. One of the first things I did was try setting it to "1" for "line buffer mode," but no matter what I set it to nothing actually changes.